src/core/currency/wallet/currency-wallet-api.js

```diff
@@ -459,6 +460,23 @@ export function makeCurrencyWalletApi(
       return getMax('0', add(balance, '1'))
     },
 
+    get stakingSettings(): EdgeStakingSettings {
+      return engine.stakingSettings != null
+        ? engine.stakingSettings
+        : { stakingEnabled: false }
+    },
+
+    async changeStakingSettings(
+      stakingSettings: EdgeStakingSettings
+    ): Promise<EdgeTransaction> {
+      if (engine.changeStakingSettings == null) {
+        throw new Error('This currency does not support staking')
+      }
+      const tx = await engine.changeStakingSettings(stakingSettings)
+      update(out) // Check for changes to this.stakingSettings
```

Comment:
Oops. Need to add a callback to the engine to tell us it changed.

Reply:
True